src/components/textarea/textarea.tsx

```diff
@@ -0,0 +1,498 @@
+          ref={this.setTextareaEl}
+          required={this.required}
+          rows={this.rows}
+          value={this.value}
+          wrap={this.wrap}
+        />
+        <span class="content">
+          <slot onSlotchange={this.contentSlotChangeHandler} />
+        </span>
+        {
+          <footer
+            class={{
+              [CSS.footer]: true,
+              [CSS.readonly]: this.readonly,
+              [CSS.hide]: !hasFooter
+            }}
+            key={CSS.footer}
+            ref={(el) => (this.footerEl = el as HTMLElement)}
+          >
+            <div class="slot-container">
+              <slot name={SLOTS.footerStart} onSlotchange={this.footerStartSlotChangeHandler} />
+              <slot name={SLOTS.footerEnd} onSlotchange={this.footerEndSlotChangeHandler} />
+            </div>
+            {this.renderCharacterLimit()}
+          </footer>
+        }
+        <HiddenFormInputSlot component={this} />
+      </Host>
+    );
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  //--------------------------------------------------------------------------
+
+  /** Sets focus on the component. */
+  @Method()
+  async setFocus(): Promise<void> {
+    await componentLoaded(this);
+    this.textareaEl.focus();
+  }
+
+  /** Selects all text of the component's `value`. */
+  @Method()
+  async selectText(): Promise<void> {
+    this.textareaEl.select();
+  }
+  //--------------------------------------------------------------------------
+  //
+  //  Private Properties/ State
+  //
+  //--------------------------------------------------------------------------
+
+  formEl: HTMLFormElement;
+
+  defaultValue: Textarea["value"];
+
+  labelEl: HTMLCalciteLabelElement;
+
+  textareaEl: HTMLTextAreaElement;
+
+  footerEl: HTMLElement;
+
+  @State() effectiveLocale = "";
+
+  @State() endSlotElements: Element[];
+
+  @State() startSlotElements: Element[];
+
+  @Watch("effectiveLocale")
+  effectiveLocaleChange(): void {
+    updateMessages(this, this.effectiveLocale);
+  }
+
+  @State() defaultMessages: TextareaMessages;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  //--------------------------------------------------------------------------
+
+  onFormReset(): void {
+    this.value = this.defaultValue;
+  }
+
+  onLabelClick(): void {
+    this.setFocus();
+  }
+
+  handleInput = (event: InputEvent): void => {
+    this.value = event.target["value"];
+    this.calciteTextareaInput.emit();
+  };
+
+  handleChange = (): void => {
+    this.calciteTextareaChange.emit();
+  };
+
+  footerEndSlotChangeHandler = (event: Event): void => {
+    this.endSlotElements = slotChangeGetAssignedElements(event);
+    if (this.disabled) {
+      this.disablePointerEvents(this.disabled, this.endSlotElements);
+    }
+  };
+
+  footerStartSlotChangeHandler = (event: Event): void => {
```

Comment:
Could possibly combine both footerEndSlotChangeHandler and footerStartSlotChangeHandler into one function (haven't tested this but saw a pattern here):

```
private footerSlotChangeHandler = (event: Event, position: 'end' | 'start' = 'start') => {
  const slotElements = `${position}SlotElements`;

  this[slotElements] = slotChangeGetAssignedElements(event);

  if (this.disabled) this.disablePointerEvents(this.disabled, this.[slotElements]);
};
```

Reply:
yeah we can combine them but in the new iteration these functions are no longer needed.